packages/gatsby/src/cache/cache-fs.ts

```diff
@@ -0,0 +1,268 @@
+const fs = require(`fs`)
+const crypto = require(`crypto`)
+const path = require(`path`)
+const promisify = require(`util`).promisify
+const jsonFileStore = require(`./json-file-store`)
+const wrapCallback = require(`./wrap-callback`)
+
+import reporter from "gatsby-cli/lib/reporter"
+
+// Cache read/writes are async. This locking attempts to prevent race conditions.
+// It's not ideal and locks timeout after 10s. It's mostly a port from when this
+// mechanism was using fs-based locks. Doing it in-memory reduces IO pressure and
+// we currently don't support two concurrent builds at the same time anyways.
+const globalGatsbyCacheLock = new Map()
+
+/**
+ * construction of the disk storage
+ * @param {object} [args] options of disk store
+ * @param {string} [args.path] path for cached files
+ * @param {number} [args.ttl] time to life in seconds
+ * @param {boolean} [args.zip] zip content to save diskspace
+ * @todo {number} [args.maxsize] max size in bytes on disk
+ * @param {boolean} [args.subdirs] create subdirectories
+ * @returns {DiskStore}
+ */
+exports.create = function (args): typeof DiskStore {
+  return new DiskStore(args && args.options ? args.options : args)
+}
+
+function DiskStore(options): void {
+  options = options || {}
+
+  this.options = {
+    path: options.path || `./cache` /* path for cached files  */,
+    ttl:
+      options.ttl >= 0
+        ? +options.ttl
+        : 60 * 60 /* time before expiring in seconds */,
+    maxsize: options.maxsize || Infinity /* max size in bytes on disk */,
+    subdirs: options.subdirs || false,
+    zip: options.zip || false,
+    lockFile: {
+      //check lock at 0ms 50ms 100ms ... 400ms 1400ms 1450ms... up to 10 seconds, after that just asume the lock is staled
+      wait: 400,
+      pollPeriod: 50,
+      stale: 10 * 1000,
+      retries: 10,
+      retryWait: 600,
+    },
```

Comment:
I think those options are not used anymore. We should either remove those or actually use some of those in the `_lock` implementation

Reply:
I actually didn't want to cut any deeper into this plugin in this PR than necessary.

A followup PR would also drop `zip` and some other parts. And try to refactor the thing to a proper es6 class, drop the `wrapCallback` wrappers, etc. Make sense?